core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -965,6 +965,13 @@ public Builder assignUUID(String newUuid) {
       return this;
     }
 
+    // it is only safe to set the format version directly while creating tables
+    // in all other cases, use the method below to safely upgrade the format version
+    private Builder setFormatVersion(int newFormatVersion) {
```

Comment:
The doc comment here is confusing since the method is write below here, I think you should say "in all other cases, use upgradeFormatVersion.

We also could try to encapsulate some of this doc into the method name?  newTableFormatVersion?

Reply:
I changed the method name and the doc.